src/main/com/cw/decoder/GalacticQueryDecoder.java

```diff
@@ -0,0 +1,89 @@
+package main.com.cw.decoder;
+
+import main.com.cw.converter.NumberConverter;
+import main.com.cw.domain.GalacticCode;
+import main.com.cw.domain.GalacticQuery;
+import main.com.cw.domain.Metal;
+import main.com.cw.mapper.GalacticInputMapper;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Created by cwijayasundara on 23/01/2017.
+ *
+ * This class has the logic to decode the inter galactic codes and prices and print out the responses.
+ *
+ */
+
+public class GalacticQueryDecoder {
+
+    private GalacticInputMapper galacticInputMapper;
+
+    private NumberConverter numberConverter;
+
+    private final String invalidQueryString = "I have no idea what you are talking about";
+
+    private final String credits = "Credits";
+
+    public GalacticQueryDecoder(GalacticInputMapper galacticInputMapper, NumberConverter numberConverter){
+        this.galacticInputMapper = galacticInputMapper;
+        this.numberConverter = numberConverter;
+    }
+
+    /*
+     * This method decode the galactic queries and produce the expected output.
+     */
+    public boolean decodeGalacticQueries(){
+
+        List<GalacticQuery> galacticQueries = galacticInputMapper.getQueryListFromInput();
+        Iterator<GalacticQuery> itr = galacticQueries.iterator();
+        boolean isOk = false;
+
+        while(itr.hasNext()){
+            GalacticQuery query = itr.next();
+            StringBuilder consolidatedRomanCodes=new StringBuilder();
+            StringBuilder consolidatedGalacticCodes = new StringBuilder();
+            List<GalacticCode> galacticCodes = query.getGalacticCode();
+            printOutput(query, consolidatedRomanCodes, consolidatedGalacticCodes, galacticCodes);
```

Comment:
Its probably better to do separate logic and print code. Then its easier to test the output of a logic without worrying  out asserting printlns.